app/helpers/check_answers_helper.rb

```diff
@@ -1,161 +1,218 @@
+  end
 
-      next if skip_question?(question)
+  def match_number_to_list_item(numbers, list_item)
+    value = numbers.with_indifferent_access["#{list_item.downcase.gsub(' ', '_')}_number"]
+    value.present? ? number_with_delimiter(value, delimiter: ",") : "0"
+  end
 
-      next transport_type if question.eql?("transport_type")
-      next offer_care_qualifications if question.eql?("offer_care_qualifications")
-      next offer_staff_type if question.eql?("offer_staff_type")
-      next link_to_parent_page(question, "offer_staff_type") if question.eql?("offer_staff_charge")
-      next link_to_parent_page(question, "construction_services") if question.eql?("construction_services_other")
-      next link_to_parent_page(question, "construction_services") if question.eql?("construction_cost")
-      next link_to_parent_page(question, "it_services") if question.eql?("it_services_other")
-      next link_to_parent_page(question, "it_services") if question.eql?("it_cost")
-      next link_to_parent_page(question, "transport_type") if question.eql?("transport_cost")
-      next link_to_parent_page(question, "offer_care_qualifications") if question.eql?("care_cost")
-      next link_to_parent_page(question, "rooms_number") if question.eql?("accommodation_cost")
-      next link_to_parent_page(question, "offer_space_type") if question.eql?("space_cost")
+  def staff_numbers
+    {
+      cleaners_number: session[:offer_staff_number][:cleaners_number],
+      developers_number: session[:offer_staff_number][:developers_number],
+      medical_staff_number: session[:offer_staff_number][:medical_staff_number],
+      office_staff_number: session[:offer_staff_number][:office_staff_number],
+      security_staff_number: session[:offer_staff_number][:security_staff_number],
+      trainers_or_coaches_number: session[:offer_staff_number][:trainers_number],
+      translators_number: session[:offer_staff_number][:translators_number],
+      other_staff_number: session[:offer_staff_number][:other_staff_number],
+    }.with_indifferent_access
+  end
 
-      value = concat_answer(session[question], question)
+  def space_numbers
+    {
+      other_number: session[:offer_space_type_other],
+      warehouse_space_number: session[:warehouse_space_description],
+      office_space_number: session[:office_space_description],
+    }.with_indifferent_access
+  end
 
+  def question_items(question)
+    if session[question].is_a?(Array)
       [{
-        field: t("coronavirus_form.questions.#{question}.title"),
-        value: sanitize(value),
-        edit: {
-          href: "#{question.dasherize}?change-answer",
-        },
+        field: I18n.t("coronavirus_form.questions.#{question}.title"),
+        value: session[question],
+        type: :list,
+      }]
+    else
+      [{
+        field: I18n.t("coronavirus_form.questions.#{question}.title"),
+        value: session[question],
       }]
     end
```

Comment:
I'd suggest:
```ruby
def question_items(question)
  question_items = {
    field: I18n.t("coronavirus_form.questions.#{question}.title"),
    value: session[question],
  }
  question_items[:type] = :list if session[question].is_a?(Array)
  [question_items]
end
```
Again - that makes it clear what difference the question being in a array makes to the output

Reply:
This is outdated 😺